packages/angular_devkit/schematics/src/engine/task.ts

```diff
@@ -33,73 +27,38 @@ export interface TaskExecutorFactory<T> {
   create(options?: T): Promise<TaskExecutor> | Observable<TaskExecutor>;
 }
 
-export interface TaskId {
-  readonly id: number;
-}
+export type TaskId = number & { __PRIVATE_DEVKIT_TASK_ID: void; };
 
 export interface TaskInfo {
-  readonly id: number;
-  readonly priority:  number;
+  readonly id: TaskId;
   readonly configuration: TaskConfiguration;
   readonly context: SchematicContext;
 }
 
 export class TaskScheduler {
-  private _queue = new PriorityQueue<TaskInfo>((x, y) => x.priority - y.priority);
+  private _tasks = new PartiallyOrderedSet<TaskId>();
   private _taskIds = new Map<TaskId, TaskInfo>();
   private static _taskIdCounter = 1;
 
   constructor(private _context: SchematicContext) {}
 
-  private _calculatePriority(dependencies: Set<TaskInfo>): number {
```

Comment:
I kind of liked the deceptive simplicity of the priority queue based on dependency depth.

I was considering moving it down one step in the pipeline and building a graph of all the tasks first so that they could be analyzed and potentially re-configured/arranged based on their dependencies and configuration.

Reply:
I'm okay removing this commit. `PartiallyOrderedSet` is different than the priority queue in a lot of things, and the `TaskScheduler` also had a different behaviour too; it's not a set. Running the same configuration twice would actually run it twice. In the new code it wouldn't.

Just let me know if you want me to remove it and I'll do that. I started work in the TaskScheduler to allow for schematic tasks, then decided to go with the Workflow abstraction instead.

Reply:
hmm...I'd probably go with removing it for now from this PR.  The underlying class is still there so it's just the _glue_ code here and we can add that in again moving forward.

Reply:
Done.